File: transpose/Transpose.java
import java.util.Arrays;
class Transpose {

public static void main(String[] arg){
Transpose t = new Transpose();
int[][] matrix = {{1, 2, 3}, {4, 5, 6}};
matrix = t.transpose(matrix);
System.out.println(Arrays.deepToString(matrix));
}

public int[][] transpose(int[][] A) {
        //0ms runtime. Faster than 100% on leetcode problem 863
        int[][] B = new int[A[0].length][A.length];
        for(int i = 0; i < A.length; i++){
            for(int j = 0; j < B.length; j++){ 
                B[j][i] = A[i][j];
            }
        }
        return B;
    }
}

File: transpose/transpose.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> transpose(vector<vector<int>>& matrix) {
    int size = matrix.size();
    vector<vector<int>> transpose(matrix.at(0).size(), vector<int>(size));
    for(int i = 0;  i < size; i++){
        for(int j = 0; j < matrix.at(i).size(); j++){
            transpose.at(j).at(i) = matrix.at(i).at(j);
        }
    }
    return transpose;
}

int main() {
	vector<vector<int>> matrix = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
	transpose(matrix);
}